DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                        : existingUserWithEmail.LastName,
+                    jobGroupId = existingUserWithEmail.JobGroupId == 10
+                        ? delegateAccount.JobGroupId_deprecated
+                        : existingUserWithEmail.JobGroupId,
+                    professionalRegistrationNumber =
+                        string.IsNullOrEmpty(existingUserWithEmail.ProfessionalRegistrationNumber)
+                            ? delegateAccount.ProfessionalRegistrationNumber_deprecated
+                            : existingUserWithEmail.ProfessionalRegistrationNumber,
+                    profileImage = existingUserWithEmail.ProfileImage ?? delegateAccount.ProfileImage_deprecated,
+                    active = existingUserWithEmail.Active || delegateAccount.Active,
+                    resetPasswordId = existingUserWithEmail.ResetPasswordId ??
+                                      delegateAccount.ResetPasswordId_deprecated,
+                    hasBeenPromptedForPrn = existingUserWithEmail.HasBeenPromptedForPrn ||
+                                            delegateAccount.HasBeenPromptedForPrn_deprecated,
+                    learningHubAuthId = existingUserWithEmail.LearningHubAuthId ??
+                                        delegateAccount.LearningHubAuthId_deprecated,
+                    hasDismissedLhLoginWarning = existingUserWithEmail.HasDismissedLhLoginWarning ||
+                                                 delegateAccount.HasDismissedLhLoginWarning_deprecated,
+                    detailsMatch = DoesDelegateAccountMatchExistingUser(delegateAccount, existingUserWithEmail, allJobGroupsMatch)
+                }
+            );
+
+            UpdateDelegateAccountUserIdEmailAndDetailsLastChecked(
+                connection,
+                existingUserWithEmail.Id,
+                delegateAccount.Id
+            );
+        }
+
+        private static bool DoesDelegateAccountMatchExistingUser(DelegateAccount delegateAccount, User existingUser, bool allJobGroupsMatch)
+        {
+            var firstNamesMatch = delegateAccount.FirstName_deprecated == existingUser.FirstName ||
+                                  string.IsNullOrWhiteSpace(existingUser.FirstName) ||
+                                  string.IsNullOrWhiteSpace(delegateAccount.FirstName_deprecated);
+
+            var lastNamesMatch = delegateAccount.LastName_deprecated == existingUser.LastName ||
+                                 string.IsNullOrWhiteSpace(existingUser.LastName) ||
+                                 string.IsNullOrWhiteSpace(delegateAccount.LastName_deprecated);
+
+            var prnMatch = delegateAccount.ProfessionalRegistrationNumber_deprecated == existingUser.ProfessionalRegistrationNumber ||
+                           string.IsNullOrWhiteSpace(existingUser.ProfessionalRegistrationNumber) ||
+                           string.IsNullOrWhiteSpace(delegateAccount.ProfessionalRegistrationNumber_deprecated);

```

Comment:
Do we also need to take HasBeenPromptedForPRN into account?

Reply:
No idea, @SteveJacksonSoft?

Reply:
Nah, if the user's been prompted for one of their delegate accounts then they've been prompted. So we can safely set that to true if it's true for any delegateAccount. We only need users to check details if we've made arbitrary decisions about which value to give a field when merging.